From d9f595be0f7142a8509fd4b5dd82b863d4919c21 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Thu, 13 Feb 2025 19:07:07 -0800
Subject: [PATCH 3/3] Disable hipcc passing hip-device-libs.

In modern times, the clang driver is far more knowledgable about this stuff and can operate without instruction.
---
 amd/hipcc/src/hipBin_amd.h | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/amd/hipcc/src/hipBin_amd.h b/amd/hipcc/src/hipBin_amd.h
index a37c07836620..fb3942bee5f7 100644
--- a/amd/hipcc/src/hipBin_amd.h
+++ b/amd/hipcc/src/hipBin_amd.h
@@ -864,15 +864,16 @@ void HipBinAmd::executeHipCCCmd(vector<string> argv) {
     }
   }
 
-  if (hasHIP) {
-    fs::path bitcodeFs = roccmPath;
-    bitcodeFs /= "amdgcn/bitcode";
-    if (deviceLibPath != bitcodeFs.string()) {
-      string hip_device_lib_str = " --hip-device-lib-path=\""
-                                  + deviceLibPath + "\"";
-      HIPCXXFLAGS += hip_device_lib_str;
-    }
-  }
+  // Clang knows better how to find its device libs. Let it.
+  // if (hasHIP) {
+  //   fs::path bitcodeFs = roccmPath;
+  //   bitcodeFs /= "amdgcn/bitcode";
+  //   if (deviceLibPath != bitcodeFs.string()) {
+  //     string hip_device_lib_str = " --hip-device-lib-path=\""
+  //                                 + deviceLibPath + "\"";
+  //     HIPCXXFLAGS += hip_device_lib_str;
+  //   }
+  // }
 
   // to avoid using dk linker or MSVC linker
   if (isWindows()) {
-- 
2.43.0

